signalfx/client.py

```diff
@@ -241,18 +272,366 @@ def _send(self):
     def _batch_data(self, datapoints_list):
         raise NotImplementedError('Subclasses should implement this!')
 
-    def _post(self, data, url, session=None):
+    def _post(self, data, url, session=None, timeout=None):
+        _timeout = timeout or self._timeout
         _session = session or self._ingest_session
-        if logging.getLogger().isEnabledFor(logging.DEBUG):
-            logging.debug(
-                'Raw datastream being sent: %s', pprint.pformat(data))
+        logging.debug('Raw datastream being sent: %s', pprint.pformat(data))
         try:
-            response = _session.post(url, data=data, timeout=self._timeout)
-            logging.debug('Sending to SignalFx %s (%d %s)',
-                          'succeeded' if response.ok else 'failed',
-                          response.status_code, response.text)
+            response = _session.post(url, data=data, timeout=_timeout)
         except Exception:
-            logging.exception('Posting to SignalFx failed.')
+            logging.exception('Posting data to SignalFx failed.')
+        logging.debug('Sending to SignalFx %s (%d %s)',
```

Comment:
this statement should be inside of try block. Incase of an exception, response is never assigned